src/test/java/teammates/test/cases/browsertests/AdminEmailPageUiTest.java

```diff
@@ -73,6 +73,53 @@ private void testCompose() throws Exception {
         assertTrue(isEmailComposeElementsPresent());
         emailPage.verifyStatus("Email will be sent within an hour to recipient@email.tmt");
 
+        ______TS("send email to group - invalid file type");
+
+        emailPage.clearRecipientBox();
+        emailPage.inputGroupRecipient("invalidGroupList.xlxs");
+        emailPage.verifyStatus("Group receiver list upload failed. Please try again.");
+
+        ______TS("send email to group - no subject");
+
+        emailPage.inputContent("Email Content");
+        emailPage.clearRecipientBox();
+        emailPage.clearSubjectBox();
+        emailPage.inputGroupRecipient("validGroupList.txt");
+        String groupListFileKey = emailPage.getGroupListFileKey();
+        emailPage.verifyStatus("Group receiver list successfully uploaded to Google Cloud Storage");
+        emailPage.verifyGroupListFileKey(groupListFileKey);
+        emailPage.clickSendButton();
+        assertTrue(hasStatusMessageNoSubject());
+
+        ______TS("send email to group - success");
+
+        emailPage.inputSubject("Email Subject");
+        emailPage.clickSendButton();
+        assertFalse(hasErrorMessage());
+        assertTrue(isEmailComposeElementsPresent());
+        emailPage.verifyStatus("Email will be sent within an hour to uploaded group receiver's list.");
+        emailPage.deleteGroupListFile(groupListFileKey);
+
+        ______TS("send email to groupmode and addressmode - no subject");
+
+        emailPage.inputRecipient("recipient@email.tmt");
+        emailPage.inputContent("Email Content");
+        emailPage.inputGroupRecipient("validGroupList.txt");
+        groupListFileKey = emailPage.getGroupListFileKey();
```

Comment:
Is it okay if I'm using groupListFileKey variable from test case "send email to group-no subject"
or should I create a new variable for this test case?